kernel/sched.c

```diff
@@ -1714,6 +1714,10 @@ static void end_thread(struct k_thread *thread)
 		unpend_all(&thread->join_queue);
 		update_cache(1);
 
+#if defined(CONFIG_FPU) && defined(CONFIG_FPU_SHARING)
```

Comment:
Why test both?  Surely the latter is the only one you need.

Reply:
I agree, only the latter check is needed, and also that `arch_thread_end` is a better place for this eventually